### ISSUE TITLE
Sample larger than population or is negative

### ISSUE THREAD

**Issue by thibautflottat** [InstaPy/InstaPy#5931]
Hello ! First of all, I would like to thank the dev, I am discovering this tools which looks very good. But, I cannot use it for now.

```
(bots) [thibaut@neowise instapy]$ python quickstart.py 
InstaPy Version: 0.6.12
 ._.  ._.  ._.  ._.  ._.  ._.  ._.  ._.  ._. 
Workspace in use: "/home/thibaut/InstaPy"
OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO
INFO [2020-11-29 14:28:31] [thibautflt]  Session started!
oooooooooooooooooooooooooooooooooooooooooooooooooooooo
INFO [2020-11-29 14:28:31] [thibautflt]  -- Connection Checklist [1/2] (Internet Connection Status)
INFO [2020-11-29 14:28:31] [thibautflt]  - Internet Connection Status: ok
INFO [2020-11-29 14:28:31] [thibautflt]  - Current IP is "79.87.233.10" and it's from "France/FR"
INFO [2020-11-29 14:28:31] [thibautflt]  -- Connection Checklist [2/2] (Hide Selenium Extension)
INFO [2020-11-29 14:28:31] [thibautflt]  - window.navigator.webdriver response: None
INFO [2020-11-29 14:28:31] [thibautflt]  - Hide Selenium Extension: ok
Cookie file not found, creating cookie...
................................................................
INFO [2020-11-29 14:28:49] [thibautflt]  Logged in successfully!
''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
INFO [2020-11-29 14:28:49] [thibautflt]  Saving account progress...
______________________________________________________________________________
INFO [2020-11-29 14:29:03] [thibautflt]  Starting to follow user `Followers`..
""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
INFO [2020-11-29 14:29:03] [thibautflt]  User 'les.explorateurs.auvergnats' [1/3]
INFO [2020-11-29 14:29:12] [thibautflt]  GraphQL query hash: [c76146de99bb02f6415203be841dd25a]

INFO [2020-11-29 14:29:15] [thibautflt]  Sessional Live Report:
	|> No any statistics to show

On session start was FOLLOWING 667 users & had 2976 FOLLOWERS
[Session lasted 46.36 seconds]
OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO
INFO [2020-11-29 14:29:15] [thibautflt]  Session ended!
ooooooooooooooooooooooooooooooooooooooooooooooooooooooo

Traceback (most recent call last):
  File "quickstart.py", line 39, in <module>
    bot.follow_user_followers(list, amount=16, randomize=False, interact=True)
  File "/home/thibaut/venv/bots/lib/python3.8/site-packages/instapy/instapy.py", line 3415, in follow_user_followers
    person_list, simulated_list = get_given_user_followers(
  File "/home/thibaut/venv/bots/lib/python3.8/site-packages/instapy/unfollow_util.py", line 1070, in get_given_user_followers
    person_list, simulated_list = get_users_through_dialog_with_graphql(
  File "/home/thibaut/venv/bots/lib/python3.8/site-packages/instapy/unfollow_util.py", line 888, in get_users_through_dialog_with_graphql
    followers_list = random.sample(followers_list, real_amount)
  File "/usr/lib/python3.8/random.py", line 363, in sample
    raise ValueError("Sample larger than population or is negative")
ValueError: Sample larger than population or is negative
```

I have a problem with the 'follow_user_followers' function. I'm using the last version of instapy in a venv, and I modified login_util.py as in https://github.com/timgrossmann/InstaPy/issues/5886#issuecomment-727207957
I checked other issues related to the ValueError, no success. The same occurs using docker-compose.

Do you have any idea ? :o
Thank you.

**Comment by elulcao** [comment]
Hi

this error `Sample larger than population or is negative` comes from `random.py`,
you need to check your limits in the `quickstart.py `

`followers_list = random.sample(followers_list, real_amount)` I guess something here is out of boundaries 
 
```python
File "/usr/lib/python3.8/random.py", line 363, in sample
    raise ValueError("Sample larger than population or is negative")
```

**Comment by thibautflottat** [comment]
Thank you for your answer @elulcao 

Actually, my quickstart.py is quite simple :

```
from instapy import InstaPy
from instapy.util import smart_run

# Write your automation here
# Stuck? Look at the github page https://github.com/InstaPy/instapy-quickstart

insta_username = "user"
insta_password = "password"

targets = ["user1", "user2", "user3"]

bot = InstaPy(username=insta_username, password=insta_password, headless_browser=True)
with smart_run(bot):

    # General settings
    bot.set_relationship_bounds(
        enabled=True,
        potency_ratio=None,
        delimit_by_numbers=True,
        max_followers=6000,
        max_following=3000,
        min_followers=45,
        min_following=77,
        min_posts=10
    )

    bot.set_user_interact(amount=5, randomize=True, percentage=50, media='Photo')

    #session.set_do_like(enabled=True, percentage=100)

    # Activities
    bot.follow_user_followers(targets, amount=16, randomize=False, interact=True)
```

All users in targets are defined in the limits I chose. I add some "print(followers_list, real_amount)" in unfollow_util.py, those object appears to be empty. Since this morning, the error changed (maybe instagram did something ?) and it cant load followers list :


```

(bots) [thibaut@neowise instapy]$ python quickstart.py 
InstaPy Version: 0.6.12
 ._.  ._.  ._.  ._.  ._.  ._.  ._.  ._.  ._. 
Workspace in use: "/home/thibaut/InstaPy"
OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO
INFO [2020-11-30 08:45:34] [thibautflt]  Session started!
oooooooooooooooooooooooooooooooooooooooooooooooooooooo
INFO [2020-11-30 08:45:34] [thibautflt]  -- Connection Checklist [1/2] (Internet Connection Status)
INFO [2020-11-30 08:45:35] [thibautflt]  - Internet Connection Status: ok
INFO [2020-11-30 08:45:35] [thibautflt]  - Current IP is "79.87.233.10" and it's from "France/FR"
INFO [2020-11-30 08:45:35] [thibautflt]  -- Connection Checklist [2/2] (Hide Selenium Extension)
INFO [2020-11-30 08:45:35] [thibautflt]  - window.navigator.webdriver response: None
INFO [2020-11-30 08:45:35] [thibautflt]  - Hide Selenium Extension: ok
Cookie file not found, creating cookie...
................................................................
INFO [2020-11-30 08:45:52] [thibautflt]  Logged in successfully!
''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
INFO [2020-11-30 08:45:52] [thibautflt]  Saving account progress...
INFO [2020-11-30 08:46:45] [thibautflt]  Failed to get followers count of 'b'thibautflt''  ~empty list
INFO [2020-11-30 08:47:35] [thibautflt]  Failed to get following count of 'b'thibautflt''  ~empty list
WARNING [2020-11-30 08:47:35] [thibautflt]  Unable to save account progress, skipping data update Message: TypeError: window._sharedData.entry_data.ProfilePage is undefined

______________________________________________________________________________
INFO [2020-11-30 08:47:48] [thibautflt]  Starting to follow user `Followers`..
""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
INFO [2020-11-30 08:47:48] [thibautflt]  User 'les.explorateurs.auvergnats' [1/3]
INFO [2020-11-30 08:48:43] [thibautflt]  Failed to get followers count of 'b'les.explorateurs.auvergnats''  ~empty list
INFO [2020-11-30 08:49:33] [thibautflt]  Failed to get following count of 'b'les.explorateurs.auvergnats''  ~empty list
INFO [2020-11-30 08:49:33] [thibautflt]  'les.explorateurs.auvergnats' has no followers

INFO [2020-11-30 08:49:33] [thibautflt]  Grabbed 0 usernames from 'les.explorateurs.auvergnats's `Followers` to do following

INFO [2020-11-30 08:49:33] [thibautflt]  User 'tourisminauvergne' [2/3]
INFO [2020-11-30 08:50:29] [thibautflt]  Failed to get followers count of 'b'tourisminauvergne''  ~empty list
INFO [2020-11-30 08:51:19] [thibautflt]  Failed to get following count of 'b'tourisminauvergne''  ~empty list
INFO [2020-11-30 08:51:19] [thibautflt]  'tourisminauvergne' has no followers

INFO [2020-11-30 08:51:19] [thibautflt]  Grabbed 0 usernames from 'tourisminauvergne's `Followers` to do following

INFO [2020-11-30 08:51:19] [thibautflt]  User 'destinationvolcans' [3/3]
INFO [2020-11-30 08:52:14] [thibautflt]  Failed to get followers count of 'b'destinationvolcans''  ~empty list
INFO [2020-11-30 08:53:04] [thibautflt]  Failed to get following count of 'b'destinationvolcans''  ~empty list
INFO [2020-11-30 08:53:04] [thibautflt]  'destinationvolcans' has no followers

INFO [2020-11-30 08:53:04] [thibautflt]  Grabbed 0 usernames from 'destinationvolcans's `Followers` to do following

INFO [2020-11-30 08:53:04] [thibautflt]  Finished following 3 users' `Followers`! xD

INFO [2020-11-30 08:53:04] [thibautflt]  Followed: 0
INFO [2020-11-30 08:53:04] [thibautflt]  Already followed: 0
INFO [2020-11-30 08:53:04] [thibautflt]  Not valid users: 0

INFO [2020-11-30 08:53:04] [thibautflt]  Liked: 0
INFO [2020-11-30 08:53:04] [thibautflt]  Already Liked: 0
INFO [2020-11-30 08:53:04] [thibautflt]  Commented: 0
INFO [2020-11-30 08:53:04] [thibautflt]  Inappropriate: 0

INFO [2020-11-30 08:53:04] [thibautflt]  Sessional Live Report:
	|> No any statistics to show


[Session lasted 7.52 minutes]
OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO
INFO [2020-11-30 08:53:04] [thibautflt]  Session ended!
ooooooooooooooooooooooooooooooooooooooooooooooooooooooo
```

**Comment by elulcao** [comment]
Could you try with other users, it's not clear why cannot be grabbed usernames.
also you can check the geckodriver.log in `trace` mode, to see what was queried.

**Comment by thibautflottat** [comment]
Ok, I found the thing. My problem was related to this new thing about cookies. As the fix from https://github.com/timgrossmann/InstaPy/issues/5886#issuecomment-727207957 did not work for me, I also made that change https://github.com/timgrossmann/InstaPy/issues/5886#issuecomment-731767547

But it was a bad fix as the tools continues to do its things whereas I was not logged in. 
So I apply headless=false, manually logged in for the first time, and now, it is working !
